torch/lib/c10d/test/ProcessGroupNCCLTest.cpp

```diff
@@ -172,7 +173,14 @@ class AllreduceNCCLTest : public NCCLTest {
     launchDeviceSleep();
     valueInitialization();
 
-    return pg_->allreduce(tensors_);
+    using namespace torch::autograd::profiler;
+    // Make sure enabling profile does not make any issue. Note, in single
+    // process multi-device mode we do not expect any events be populated for
+    // collective operations.
+    enableProfiler({ProfilerState::CPU});
+    auto results = pg_->allreduce(tensors_);
+    disableProfiler();
+    return results;
```

Comment:
Generally a test should assert something/some condition. Could we search through `results` and verify there is an allreduce here? You can see https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_misc.cpp#L2185-L2198 as an example.

Reply:
Can we do a single device per process test somehow? In the current version are the profiling results empty? 

Also tangentially related - can you add a comment somewhere appropriate that specifies that it only works for single process per device?

Reply:
Following up here, is it possible to add some asserts on the expected result?